src/main/java/com/github/mc1arke/sonarqube/plugin/CommunityBranchPlugin.java

```diff
@@ -67,8 +71,80 @@ public void load(CoreExtension.Context context) {
+                        .name("Base image URL")
+                        .defaultValue("https://raw.githubusercontent.com/mc1arke/sonarqube-community-branch-plugin/pr-decoration/src/main/resources/pr-decoration-images")
+                        .description("Default: https://raw.githubusercontent.com/mc1arke/sonarqube-community-branch-plugin/master/src/main/resources/pr-decoration-images")
+                        .type(PropertyType.STRING)
+                        .build(),
+
+                PropertyDefinition.builder("sonar.pullrequest.summary.comment.enabled")
+                        .category(PULL_REQUEST_CATEGORY_LABEL)
+                        .subCategory(GENERAL)
+                        .onlyOnQualifiers(Qualifiers.PROJECT)
+                        .name("Enable summary comment")
+                        .description("This enables the summary comment (if implemented).")
+                        .type(PropertyType.BOOLEAN)
+                        .defaultValue("true")
+                        .build(),
+
+                PropertyDefinition.builder("sonar.pullrequest.file.comment.enabled")
+                        .category(PULL_REQUEST_CATEGORY_LABEL)
+                        .subCategory(GENERAL)
+                        .onlyOnQualifiers(Qualifiers.PROJECT)
+                        .name("Enable file comment")
+                        .description("This enables commenting (if implemented).")
+                        .type(PropertyType.BOOLEAN)
+                        .defaultValue("true")
+                        .build(),
+
+                PropertyDefinition.builder("sonar.pullrequest.delete.comments.enabled")
+                        .category(PULL_REQUEST_CATEGORY_LABEL)
+                        .subCategory(GENERAL)
+                        .onlyOnQualifiers(Qualifiers.PROJECT)
+                        .name("Enable deleting comments")
+                        .description("This cleans up the comments from previous runs (if implemented).")
+                        .type(PropertyType.BOOLEAN)
+                        .defaultValue("false")
+                        .build(),
+
+                PropertyDefinition.builder("sonar.pullrequest.gitlab.url")
+                        .category(PULL_REQUEST_CATEGORY_LABEL)
+                        .subCategory(GITLAB_INTEGRATION_SUBCATEGORY_LABEL)
+                        .onQualifiers(Qualifiers.PROJECT)
+                        .name("URL for Gitlab (Server or Cloud) instance")
+                        .description("Example: https://ci-server.local/gitlab")
+                        .type(PropertyType.STRING)
+                        .defaultValue("https://gitlab.com")
+                        .build(),
+
+                PropertyDefinition.builder("sonar.pullrequest.gitlab.token")
+                        .category(PULL_REQUEST_CATEGORY_LABEL)
+                        .subCategory(GITLAB_INTEGRATION_SUBCATEGORY_LABEL)
+                        .onQualifiers(Qualifiers.PROJECT)
```

Comment:
Should this be global across all projects, or set on each project?
```suggestion
                        .onQualifiers(Qualifiers.APP)
```

Reply:
this needs to be per project, since each project might be connected to a different gitlab instance or use a differnt gitlab user